Add support for non-multitouch v1_01 PCBs

ian-thompson-yv opened the pull request:
## Description

The multitouch branches are very jumpy and hard to use on MacOS. The basic `peacock` branch works much more smoothly for cursor, scrolling, tap and two finger tap. This PR adds support for version 1_01+ PCBs with the updated traces and working encoders.

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

george-norton committed:
Initial commit of 'georgen/peacock'

# Conflicts:
#	keyboards/georgen/peacock/config.h
#	keyboards/georgen/peacock/info.json
#	keyboards/georgen/peacock/keymaps/default/keymap.c
#	keyboards/georgen/peacock/readme.md
#	keyboards/georgen/peacock/rules.mk
#	quantum/pointing_device/pointing_device.c

george-norton committed:
Attempt to reduce some jitter when holding your finger down.

george-norton committed:
Tweaked a few gesture thresholds. Made more stuff user configurable. Suppress log messages for fingers 2-10.

george-norton committed:
Brought across a few changes from the multitouch_experiments branch.

george-norton committed:
Rebase pavonis+peacock on develop

george-norton committed:
Buzz the haptics to indicate we are rebooting to bootloader.

george-norton committed:
Added maxtouch debug interface

george-norton committed:
Fix stylus regression since rebase.

george-norton committed:
Missing t47 registers

george-norton committed:
Reduce gain to 10.

ian-thompson-yv committed:
keyboards/georgen/peacock: Add support for v1_01 board without multitouch

Add support for v1_01 peacock boards on the non-multitouch branch. This allows
the trackpads to work much more cleanly and smoothly on MacOS where multitouch
is buggy.

ian-thompson-yv committed:
keyboards/georgen/peacock/readme.md: Update the readme for peacock trackpads

ian-thompson-yv committed:
fixup! keyboards/georgen/peacock: Add support for v1_01 board without multitouch

george-norton committed:
Current test settings.

george-norton committed:
New sensor tuning, and more mouse mode gestures.

george-norton committed:
Mouse mode debug commands.

george-norton commented:
Hello, sorry I just spotted this PR. I have been making some improvements for MacOS support, can you retry the multitouch_experiment branch.

The latest code should also add 3 finger gesture support for mouse emulation mode (Tap for middle click, Swipe up for L_GUI, swipe left/right for back/forward).

Commit:
Configuration changes based on some experiments and reading the damn protocol guide.

T7:
- Let the analog acquisition free-run to maximize speed. No delays between samples at all FOR MAXIMUM PERFORRRRRMANCE.
T100:
- SCANEN - enable close scanning to help detect multiple fingers
- Reduced MXT_CPI to 400; avoids exceeding cfg.xsize and cfg.ysize specs in T100 object (limited to 65,534 per page 55 of the MaxTouch protocol guide).
T42:
- Activated the T42 touch suppression object and picked some sane-seeming defaults for the various settings.
T46:
- Activate the inrush current resistors, set to 1k.
T47:
- Tune CONTMAX down a little, so that fingers are less likely to be mistaken as stylus touches.
- Adjusted some of the other parameters with testing.

george-norton committed:
Remove the CPI handling.

Commit:
Added code to activate the T56 and T65 objects, which purportedly help with noise.

george-norton committed:
Fix peacock build warning

george-norton committed:
Code cleanup.

ian-thompson-yv commented:
> Hello, sorry I just spotted this PR. I have been making some improvements for MacOS support, can you retry the multitouch_experiment branch.
> 
> The latest code should also add 3 finger gesture support for mouse emulation mode (Tap for middle click, Swipe up for L_GUI, swipe left/right for back/forward).

Do I need to enable some flag for mouse emulation mode specifically. The branch is a lot less jittery now and usable although the scroll is very twitchy. The gestures aren't registering.

george-norton committed:
Fix x/y range on peacock.

george-norton commented:
Hello, sorry I just ran a test and noticed that I had broken the resolution configuration. Movement vertically was too fast, I have pushed a fix for that. I don't have a Mac, but I have forced mouse mode on Linux and I am getting gestures.

Sorry, the branch is in a bit of flux at the moment, there is another trackpad in the works and we are running a beta test, we are attempting to tune things a little better.

Commit:
Activated T80 and tweaked some other settings based on tester feedback.

george-norton committed:
Add T25

george-norton committed:
Improve test reporting.

Commit:
Made a bunch of configuration changes and tweaks to the sensor tuning to improve perfrormance in situations where the trackpad is connected to a computer with a lousy power supply.

Commit:
Small changes to improve tuning. WIP.

Commit:
More miscellaneous tuning...

george-norton committed:
Fix building digitzer with no hardware.

george-norton committed:
Fix compilation of chibos targets without digitizers.

george-norton committed:
Fix lint warnings.

george-norton committed:
Fixup digitizer onekey a little.

george-norton committed:
Procyon support.

george-norton committed:
Fix build issues.

george-norton committed:
Procyon tuning tweaks

george-norton committed:
Fixup tuning for all boards

george-norton committed:
Fix 100Hz Procyon report rate limit.

george-norton committed:
Refactoring tip handling.

george-norton committed:
Initial Dilemma Proycon support.

george-norton committed:
Fix tuning for 2 finger events.

george-norton committed:
Fix rotation in digitizer mode

george-norton committed:
Push pop physical state

george-norton committed:
Increase the shared ep size if digitizer is enabled.

george-norton committed:
Don't store user modified digitizer state.

george-norton committed:
Remove digitizer mousekey handling

george-norton committed:
Allow the stylus to be enabled if required.

george-norton committed:
Enable free running on peacock.

george-norton committed:
Stylus tweaks

george-norton committed:
Handle Un suppressed up.

george-norton committed:
*Azoteq fixes from dasky,

george-norton committed:
CPI handling for the digitizer pointing device driver.

george-norton committed:
Fixes for stuck fingers.

george-norton committed:
Increase default mouse CPI

george-norton committed:
Fix onekey.

george-norton committed:
Slow down digitizer mouse scroll speed.

george-norton committed:
Tweak peacock tuning.

george-norton committed:
Add procyon r2 support.

ploopyco committed:
updated active and idle syncs per x

george-norton committed:
New Procyon

george-norton committed:
Formatting and stylus tweaks.

george-norton committed:
No digitizer fingers if no hardware.

george-norton committed:
First pass at docs update

george-norton committed:
Update some comments.

george-norton committed:
Lint the digitizer feature.

george-norton committed:
Update dilemma Procyon definition.

george-norton committed:
Stylus tweaks.

george-norton committed:
Fix dilemma rotation.

george-norton committed:
Stylus tweaks

george-norton committed:
Set peacock default folder to build v1_01

george-norton committed:
Fixes for new i2c interface.

burkfers committed:
initial release of dilemma_3x5_3_procyon

george-norton committed:
Tweaks to mouse mode

george-norton committed:
Refactory mouse fallback.

george-norton committed:
Mouse fallback improvements.

george-norton committed:
Remove logging and increase the strength of the dejittering filter.

george-norton committed:
Fix compile error.

george-norton committed:
Add DIGITIZER_REPORT_TAPS_AS_CLICKS option.

george-norton committed:
Fixup macro

george-norton committed:
Fix regression on Windows.

george-norton committed:
Add workaround for mouse mode regression.

george-norton committed:
Tweak mouse fallback.

george-norton committed:
Dont swipe unless we are reporting mouse events.

george-norton committed:
Updated docs and made the swipe gestures configurable.

george-norton committed:
Get the digitizer onto its own endpoint.

george-norton committed:
Fix report handling on digitizer endpoint.

george-norton committed:
Cleanup code.

george-norton committed:
Fix onekey

george-norton committed:
Code cleanup and fixes for onekey.

george-norton committed:
Fix digitizer endpoint size.

george-norton committed:
Allow more contacts to be captured than are reported.

george-norton committed:
Minor cleanups.

george-norton committed:
Fix onekey.

george-norton committed:
Lint changes

george-norton committed:
Propagate buttons from user callback.

george-norton committed:
Fixup comments.

george-norton committed:
Fix CI failure.

george-norton committed:
Fix lint issues.

george-norton committed:
Resync submodules

george-norton committed:
include string.h, the missing include causes a few CI test targets to fail to build.

george-norton committed:
Move digitizer finger descriptor macro

george-norton committed:
Remove some dead code.

george-norton committed:
Do not pass the digitizer_state by value, it might be big.

george-norton committed:
Tweak endpoints

george-norton committed:
Replace contact count with finger count.

george-norton committed:
Tweak the T42 touch suppression config if we have configured a high contact count.

george-norton committed:
Fix behaviour on the slave half.

getreuer committed:
[Core] Enhance Flow Tap to work better for rolls over multiple tap-hold keys. (#25200)

* Flow Tap revision for rolling press.

* Remove debugging cruft.

* Formatting fix.

zvecr committed:
amptrics/0422 - Prevent OOB in `update_leds_for_layer` (#25209)

takashicompany committed:
[Keyboard] Add Gravity-45(#25206)

* add gravity-45

* readme.md

* fix readme

* Update keyboards/green_keys/gravity_45/keyboard.json

Co-authored-by: jack <0x6a73@protonmail.com>

* run qmk format-json -i keyboards/green_keys/gravity_45/keyboard.json

* add url

* Update keyboards/green_keys/gravity_45/keyboard.json

Co-authored-by: Duncan Sutherland <dunk2k_2000@hotmail.com>

* Update keyboard.json

* Update keyboard.json

* Update keyboards/green_keys/gravity_45/keyboard.json

Co-authored-by: Drashna Jaelre <drashna@live.com>

---------

Co-authored-by: jack <0x6a73@protonmail.com>
Co-authored-by: Duncan Sutherland <dunk2k_2000@hotmail.com>
Co-authored-by: Drashna Jaelre <drashna@live.com>

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

zvecr committed:
Remove redundant keyboard headers (#25208)

chrishoage committed:
Fix Spleeb compile when pointing device is enabled (#25016)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

tzarc committed:
[Bug] Minimise force-included files (#25194)

drashna committed:
Add additional hooks for Community modules (#25050)

zvecr committed:
Workaround for resolving keyboard alias for config file values (#25228)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

zvecr committed:
Generate versions to keycode headers (#25219)

zvecr committed:
Resolve alias for `qmk new-keymap` keyboard prompts (#25210)

silvinor committed:
[Keyboard] Add Binepad KN01 (#25224)

* Add Binepad NeoKnob KN01

* post @waffle87 recommendations

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

zvecr committed:
Add battery changed callbacks (#25207)

zvecr committed:
Ensure `qmk_userspace_paths` maintains detected order (#25204)

zvecr committed:
Bind Bluetooth driver to `host_driver_t` (#25199)

tzarc committed:
Deprecate `qmk generate-compilation-database`. (#25237)

zvecr committed:
Remove force disable of NKRO when Bluetooth enabled (#25201)

yiancar committed:
Keycult 60 (#25213)

Co-authored-by: Duncan Sutherland <dunk2k_2000@hotmail.com>
Co-authored-by: jack <jack@pngu.org>

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

silvinor committed:
[Keyboard] Add Binepad KnobX1 (#25222)

Co-authored-by: Drashna Jaelre <drashna@live.com>

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

drashna committed:
[Keyboard] Update Tractyl Manuform and add F405 (weact) variant (#24764)

dunk2k committed:
Layout corrections: Zed60 (#25003)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

dabstractor committed:
Fixed print statement after enabling 32-bit layers (#25027)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

veypatch committed:
Fix Aurora sweep default keymap configuration (#25148)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

nathanscain committed:
Docs update for installing qmk with uv (#24995)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

muge committed:
CXT Studio 12E3: Fix encoder resolutions not applying (#25242)

* add resolution to encoders so they apply

* Tweak default keymap

* replace KC_UNDO with C(KC_Z) as well

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

zvecr committed:
Add debounce to duplicated defaults check (#25246)

drashna committed:
[Docs] Fix typos introduced by PR #25050 (#25250)

It isn't a drashna PR if there aren't some typos in it somewhere.

HorrorTroll committed:
Allow LVGL onekey keymap to be able compile for other board (#25005)

cyxae committed:
[Keyboard] Add Jason Hazel’s Bad Wings v2 (#25252)

Co-authored-by: Florent Allard <florent.allard@savoirfairelinux.com>

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

zvecr committed:
Add raw_hid support to host driver (#25255)

zvecr committed:
Fix Wear Leveling compilation (#25254)

cantinalabs committed:
[New Feature/Core] New RGB Matrix Animation "Starlight Smooth" (#25203)

getreuer committed:
Enable community modules to define LED matrix and RGB matrix effects. (#25187)

Co-authored-by: Joel Challis <git@zvecr.com>

qvr committed:
Fix OS_DETECTION_KEYBOARD_RESET (#25015)

Co-authored-by: Nick Brassel <nick@tzarc.org>

ingvardm committed:
Fixes the numlock indicator for Magic Force MF17 numpad (#25260)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

dlip committed:
Add Harite v2 keyboard (#24975)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

zvecr committed:
dlip/haritev2 - Post merge fixes (#25264)

zvecr committed:
Remove more USB only branches from NKRO handling (#25263)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

zvecr committed:
Deprecate `usb.force_nkro`/`FORCE_NKRO` (#25262)

nooges committed:
Add BDN9 Rev. 3 (#25261)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

zvecr committed:
Remove duplicate of SPI default config from keyboards (#25266)

zvecr committed:
Resolve miscellaneous keyboard lint warnings (#25268)

diffrentGuesser committed:
Add Zeropad (#24737)

Co-authored-by: Drashna Jaelre <drashna@live.com>
Co-authored-by: Joel Challis <git@zvecr.com>

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

zvecr committed:
Add Waveshare RP2040-Keyboard-3 support (#25269)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

zvecr committed:
Update PR checklist notes on custom matrix (#25240)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

flinguenheld committed:
Chew folders (#24785)

tzarc committed:
gcc15 AVR compilation fixes (#25238)

KarlK90 committed:
[Core] STM32G0x1 support (#24301)

tzarc committed:
Use relative paths for schemas, instead of $id. Enables VScode validation. (#25251)

clownfish-og committed:
add doio/kb03 (#24815)

Co-authored-by: Drashna Jaelre <drashna@live.com>
Co-authored-by: Joel Challis <git@zvecr.com>

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

ivndbt committed:
Move rookiebwoy to ivndbt (#25142)

KarlK90 committed:
[Chore] use {rgblight,rgb_matrix}_hsv_to_rgb overrides (#25271)

tzarc committed:
Remove outdated `nix` support due to bit-rot. (#25280)

elpekenin committed:
Add `compiler_support.h` (#25274)

zvecr committed:
Configure boards to use development_board - 0-9 (#25287)

KarlK90 committed:
[Fix] lib8tion: enable fixed scale8 and blend functions (#25272)

lib8tion: enable fixed scale8 and blend functions

These FastLED derived lib8tion functions have been fixed and enabled by
default in FastLED. QMK just never set these defines, there is no reason
to keep the buggy implementation. It is assumed that nobody relied on
the buggy behavior.

zvecr committed:
Configure boards to use development_board - UVWXYZ (#25288)

zvecr committed:
[Docs] Fix tap_hold code blocks (#25298)

zvecr committed:
salicylic_acid3/getta25 - Fix oled keymap (#25295)

zvecr committed:
Configure boards to use development_board - S (#25293)

zvecr committed:
Configure boards to use development_board - T (#25294)

zvecr committed:
2025 Q2 changelog (#25297)

Co-authored-by: Drashna Jaelre <drashna@live.com>
Co-authored-by: Nick Brassel <nick@tzarc.org>

tzarc committed:
Merge remote-tracking branch 'upstream/develop'

tzarc committed:
Branch point for 2025-05-25 Breaking Change

daskygit committed:
Fix dynamic keymap macro only sending first. (#25309)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

dependabot[bot] committed:
Bump vite from 5.4.18 to 5.4.19 in /builddefs/docsgen (#25306)

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 5.4.18 to 5.4.19.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v5.4.19/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.4.19/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-version: 5.4.19
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

RSATom committed:
Fix reference to wrong layer when LM used with right modifiers (#25303)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

idkau committed:
Fix Sofle OLED layer indication code (#25311)

Fixed oled raise/lower layer order and added adjust layer text to oled display

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

zvecr committed:
Fix typo in Battery Driver docs (#25312)

* Fix typo in Battery Driver docs

* Update battery.md

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

stephenostermiller committed:
Fix quantum.c keycode handling (#25322)

zvecr committed:
Configure boards to use development_board - P (#25317)

zvecr committed:
Configure boards to use development_board - R (#25316)

daskygit committed:
Fix dynamic keymap static assert (#25327)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

zvecr committed:
Relocate remaining `process_record_quantum` keycodes (#25328)

zvecr committed:
Remove `process_action_kb` callback (#25331)

zvecr committed:
[Docs] Fix PS/2 Driver headers (#25332)

obosob committed:
Add CrossDIY to license_violations.md (#25324)

Co-authored-by: jack <jack@pngu.org>

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

tzarc committed:
More Yunzii notes regarding refusal to provide sources. (#25335)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

bytetinkerer committed:
add kinesis/kint32 (#25013)

Co-authored-by: Drashna Jaelre <drashna@live.com>

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

spacehangover committed:
Added Encoder support for Soyuz (#25279)

zvecr committed:
Configure boards to use development_board - NO (#25338)

tzarc committed:
[tzarc/djinn] Include community modules in RGB naming. (#25343)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

tzarc committed:
Add `{rgb|led}_matrix_get_mode_name()`. (#25344)

zvecr committed:
maple_computing/launchpad - Remove broken `default_rgb` keymap (#25342)

zvecr committed:
Configure boards to use development_board - LM (#25341)

veypatch committed:
[Keyboard] Add splitkb.com's Halcyon Lily58 rev2 (#25144)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

veypatch committed:
[Keyboard] Add splitkb.com's Halcyon Ferris rev1 (#25145)

Co-authored-by: Thomas Baart <support@splitkb.com>

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

veypatch committed:
[Keyboard] Add splitkb.com's Halcyon Corne rev2 (#25143)

Co-authored-by: Dasky <32983009+daskygit@users.noreply.github.com>

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

zvecr committed:
Remove `DEFAULT_FOLDER` handling (#23281)

elpekenin committed:
[Bugfix] User eeconfig utility macros (#25350)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

clownfish-og committed:
update winry25 VID and PID (#25351)

zvecr committed:
Migrate remaining `DEFAULT_FOLDER` to keyboard aliases (#25291)

tzarc committed:
Specify language standard when generating compilation database. (#25354)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

Hardware7253 committed:
Add keyboard Lily58 Koca (#24847)

Co-authored-by: Ryan <fauxpark@gmail.com>
Co-authored-by: Drashna Jaelre <drashna@live.com>
Co-authored-by: Dasky <32983009+daskygit@users.noreply.github.com>

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

clownfish-og committed:
fix winry25tc lightsout keymap (#25353)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

Gondolindrim committed:
Add support for Protagonist PCBs (#22918)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

luroc committed:
[Docs] Fix example code on key_overrides.md (#25367)

The description of the code snippet says right alt, but the snippet itself contained the right control keycode

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

daskygit committed:
Add core handling for pointing device failures. (#25315)

tzarc committed:
More compiledb fixes. (#25355)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

zvecr committed:
Configure boards to use development_board - FGHIJ (#25370)

zvecr committed:
Configure boards to use development_board - DE (#25369)

Xelus22 committed:
[Bug] Fix `usb_endpoint_interface_lut` multiple def compile err (#25378)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

george-norton committed:
Resync submodules

george-norton committed:
Reverted changes to dilemma 3x5_3
